docs/execution-providers/CoreML-ExecutionProvider.md

```diff
@@ -49,18 +60,64 @@ Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_CoreML(so, coreml_fla
 Ort::Session session(env, model_path, so);
 ```
 
-## Configuration Options
+## Configuration Options (NEW API)
 
 There are several run time options available for the CoreML EP.
 
 To use the CoreML EP run time options, create an unsigned integer representing the options, and set each individual option by using the bitwise OR operator.
 
+ProviderOptions can be set by passing the unsigned integer to the `AppendExecutionProvider` method.
+```c++
+std::unordered_map<std::string, std::string> provider_options;
+provider_options["ModelFormat"] = std::to_string("MLProgram");
+provider_options["MLComputeUnits"] = std::to_string("ALL");
+provider_options["RequireStaticInputShapes"] = std::to_string("0");
+provider_options["EnableOnSubgraphs"] = std::to_string("0");
+```
+
+Python inference example code to use the CoreML EP run time options:
+```python
+import onnxruntime as ort
+model_path = "model.onnx"
+providers = [
+    ('CoreMLExecutionProvider', {
+        "ModelFormat": "MLProgram", "MLComputeUnits": "ALL", 
+        "RequireStaticInputShapes": "0", "EnableOnSubgraphs": "0"
+    }),
+]
+
+session = ort.InferenceSession(model_path, providers=providers)
+outputs = ort_sess.run(None, input_feed)
+```
+
+### Available Options (New API)
```

Comment:
nit: can we list the defaults for all options so a user doesn't feel the need to specify everything manually?

Reply:
Sounds good.